view/src/components/produce.js

```diff
@@ -1,617 +1,723 @@
+    color: theme.palette.grey[500],
+  },
+  search: {
+    position: "relative",
+    borderRadius: theme.shape.borderRadius,
+    marginLeft: 0,
+    width: "100%",
+  },
+  searchIcon: {
+    padding: theme.spacing(0, 2),
+    height: "100%",
+    position: "absolute",
+    pointerEvents: "none",
+    display: "flex",
+    alignItems: "center",
+    justifyContent: "center",
+  },
+  inputRoot: {
+    color: "inherit",
+  },
+  inputInput: {
+    padding: theme.spacing(1, 1, 1, 0),
+    paddingLeft: `calc(1em + ${theme.spacing(4)}px)`,
+    width: "100%",
+  },
 });
 
 const Transition = React.forwardRef(function Transition(props, ref) {
-	return <Slide direction="up" ref={ref} {...props} />;
+  return <Slide direction="up" ref={ref} {...props} />;
 });
 
 class produce extends Component {
-	constructor(props) {
-		super(props);
+  constructor(props) {
+    super(props);
 
-		this.state = {
-			produceObjects: '',
-			name: '',
-			produceId: '',
-			shippingPresetTemperature: '',
-			shippingMaintenanceTemperatureLow: '',
-			shippingMaintenanceTemperatureHigh: '',
-			price: '',
-			pricePaid: '',
-			amountMoved: '',
-			errors: [],
-			open: false,
-			uiLoading: true,
-			buttonType: '',
-			viewOpen: false
-		};
+    this.state = {
+      data: "",
+      name: "",
+      produceId: "",
+      shippingPresetTemperature: "",
+      shippingMaintenanceTemperatureLow: "",
+      shippingMaintenanceTemperatureHigh: "",
+      price: "",
+      pricePaid: "",
+      amountMoved: "",
+      errors: [],
+      open: false,
+      uiLoading: true,
+      value: "",
+      buttonType: "",
+      viewOpen: false,
+    };
 
-		this.deleteTodoHandler = this.deleteTodoHandler.bind(this);
-		this.handleEditClickOpen = this.handleEditClickOpen.bind(this);
-		this.handleViewOpen = this.handleViewOpen.bind(this);
-	}
+    this.deleteTodoHandler = this.deleteTodoHandler.bind(this);
+    this.handleEditClickOpen = this.handleEditClickOpen.bind(this);
+    this.handleViewOpen = this.handleViewOpen.bind(this);
+  }
 
-	handleChange = (event) => {
-		this.setState({
-			[event.target.name]: event.target.value
-		});
-	};
+  handleChange = (event) => {
+    this.setState({
+      [event.target.name]: event.target.value,
+    });
+  };
 
-	componentWillMount = () => {
-		authMiddleWare(this.props.history);
-		const authToken = localStorage.getItem('AuthToken');
-		axios.defaults.headers.common = { Authorization: `${authToken}` };
-		axios
-			.get('/produce')
-			.then((response) => {
-				this.setState({
-					produceObjects: response.data,
-					uiLoading: false
-				});
-			})
-			.catch((err) => {
-				console.log(err);
-			});
-	};
+  componentWillMount = () => {
+    authMiddleWare(this.props.history);
+    const authToken = localStorage.getItem("AuthToken");
+    axios.defaults.headers.common = { Authorization: `${authToken}` };
+    axios
+      .get("/produce")
+      .then((response) => {
+        this.setState({
+          data: response.data,
```

Comment:
Maybe add some documentation explaining what data is since its a super broad name (ie. contains all of the produce objects)

Reply:
Good catch thanks! Btw if anyone ends up making changes to the filtering later on, I had found that using the variable names data and value were necessary for the search library (I'll add this info to our handoff doc right now)